returnn/tf/layers/basic.py

```diff
@@ -4792,6 +4806,12 @@ def __init__(self, kind="equal", value=None, **kwargs):
       if r_last is not None:
         r = tf.logical_and(r_last, r)
       r_last = r
+    if value is not None:
+      r = op(x, value)
+      if r_last is not None:
+        r = tf.logical_and(r_last, r)
+      else:
+        r_last = r
```

Comment:
You want to change the behavior that `value` is checked last?
But does this make sense? Is this how you would expect it? (@JackTemaki What do you think?)
Do you actually have multiple sources?
I am thinking whether we should maybe only allow multiple sources + value if `kind=="equal"`, or otherwise only a single source if value is given as well.

Reply:
I replied in the other thread, this is just to propose a unified behavior, if we can find it. 
For sure it is not finished as we need to assign x2 to x after each comparison to achieve that behavior. So far it is still buggy. But we need to agree on the behavior first.

Reply:
> I am thinking whether we should maybe only allow multiple sources + value if `kind=="equal"`, or otherwise only a single source if value is given as well.

This should be good.

Reply:
I realized now that even for a single source, this is kind of arbitrary. See the discussion on the commit comment.
Esp consider this:
`{"class": "compare", "kind": "greater", "from": "x", "value": 0}`
vs:
`{"class": "compare", "kind": "greater", "value": 0, "from": "x"}`